Cargo.toml

```diff
@@ -8,4 +8,10 @@ edition = "2018"
 serde = "~1.0.27"
 serde_derive = "~1.0.27"
 serde_json = "~1.0.9"
-failure = "~0.1.1"
\ No newline at end of file
+failure = "~0.1.1"
+kafka = "0.7.0"
```

Comment:
Prefer we do not have direct dependency on Kafka, and define a trait for queuing.

Reply:
Ok

Reply:
@amitu i need some clarification on this trait. If we define new trait say Queue with queue() method in it, each struct can implement its own way of queuing. then how would we write a worker which reads those queues and stores the data to the final storage